core/src/main/scala/kafka/log/Log.scala

```diff
@@ -777,7 +778,12 @@ class Log(val dir: File,
   /**
    * The active segment that is currently taking appends
    */
-  def activeSegment = segments.lastEntry.getValue
+  def activeSegment: LogSegment = {
```

Comment:
I don't think the check in `activeSegment` isn't necessary because `segments` is never empty. `loadSegments` is called right after `val segments` in the constructor. Perhaps `loadSegments` should return segments to make it clear.